docappendertest/docappendertest.go

```diff
@@ -44,7 +45,10 @@ import (
 const TimestampFormat = "2006-01-02T15:04:05.000Z07:00"
 
 // DecodeBulkRequest decodes a /_bulk request's body, returning the decoded documents and a response body.
-func DecodeBulkRequest(r *http.Request) ([][]byte, esutil.BulkIndexerResponse) {
+func DecodeBulkRequest(r *http.Request) (
```

Comment:
> move countReader out of here and wrap the r.Body before passing it to DecodeBulkRequest. We can directly count the size in the caller rather than here.

If we wrapped from the caller, the count would be the compressed bytes read by the gzip reader in case the compression was enabled so we won't able to track the uncompressed bytes. I ended up adding a new method to return the RequestStats struct.

Reply:
Ah good point!